From 70a7e7eb9291c962a5d5e64d715e18b615c2cb08 Mon Sep 17 00:00:00 2001
From: Loic Poulain <loic.poulain@intel.com>
Date: Mon, 15 Sep 2014 16:51:06 +0200
Subject: [PATCH 1/3] net: rfkill: gpio: Fix clock status

Clock is disabled when the device is blocked.
So, clock_enabled is the logical negation of "blocked".

Change-Id: Ibb89d28e2a965ed14dcee52e74f2db2d721a7d1b
Signed-off-by: Loic Poulain <loic.poulain@intel.com>
---
 net/rfkill/rfkill-gpio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/rfkill/rfkill-gpio.c b/net/rfkill/rfkill-gpio.c
index e7e9a95..81e0aae 100644
--- a/net/rfkill/rfkill-gpio.c
+++ b/net/rfkill/rfkill-gpio.c
@@ -65,7 +65,7 @@ static int rfkill_gpio_set_power(void *data, bool blocked)
 		rfkill_set_gpio(rfkill->shutdown_gpio, 1);
 	}
 
-	rfkill->clk_enabled = blocked;
+	rfkill->clk_enabled = !blocked;
 
 	return 0;
 }
-- 
1.9.1

